plugins/highlight-portal-history.js

```diff
@@ -0,0 +1,33 @@
+// @author         Johtaja
+// @name           Highlight portals based on history
+// @category       Highlighter
+// @version        0.1.0
+// @description    Use the portal fill color to denote the portal has been visited, caputured or scouted
+
+
+// use own namespace for plugin
+window.plugin.portalHighlighterPortalsHistory = function() {};
+
+window.plugin.portalHighlighterPortalsHistory.visited = function(data) {
+  let visited = (data.portal.options.ent[2][18] & 3);
+  if (visited !== undefined) {
+    if (visited === 1) {
+      var opacity = .6;
+      data.portal.setStyle(
+        {fillColor: 'yellow' 
+        }
+      );
+    }
+    if (visited === 2 || visited ===3) {
+      var opacity = .6;
```

Comment:
The way the visited variable is being used, could consider rewriting this section as a switch/case instead.